Sprint 10 solution

Dvoychenkov opened the pull request.

Dvoychenkov committed:
- Добавлены модели данных;

Dvoychenkov committed:
- Добавлен базовый код контролеров для моделей;

Dvoychenkov committed:
- Добавлена валидация для rest-методов;

Dvoychenkov committed:
- Добавлено логирование;

Dvoychenkov committed:
- Убрали лишние зависимости из помника;
- Поправил типы полей для моделей и валидацию в контроллерах;
- Добавил тесты для валидации моделей;
- Скорректировал текста ошибок валидации;

Dvoychenkov committed:
- Убрал не совсем корректную валидацию согласно REST парадигме (PUT обновляет всю сущность, а вот PATCH — частично);

Dvoychenkov committed:
- Добавил плагин для чек-стайлинга;
- Заменил явное создание логгера на аннотацию;

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,12 +1,14 @@
+
+@Data
 public class Film {
+    Long id;
```
Не указаны модификаторы доступа

Dvoychenkov replied:
Поправил

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,14 @@
+
+@Data
+public class User {
+    Long id;
```
Не указаны модификаторы доступа

Dvoychenkov replied:
Поправил

Dvoychenkov committed:
- Добавил модификаторы доступа для классов моделей;

Dvoychenkov committed:
Дополнительное задание:
- Добавлена зависимость Soring boot validation (для валидации моделей аннотациями);
- Добавлен кастомный валидатор для даты релиза фильма
- Поправлено описание валидации моделей при помощи аннотаций;
- Убрана валидация из контролеров;
- Удалены неактуальные тесты для моделей, добавлены актуальные;
